storage/rocksdb/ha_rocksdb.cc

```diff
@@ -12518,16 +12519,43 @@ int rocksdb_check_bulk_load(THD *const thd, struct st_mysql_sys_var *var
   return 0;
 }
 
-void rocksdb_set_bulk_load_allow_unsorted(
-    THD *const thd,
-    struct st_mysql_sys_var *const var MY_ATTRIBUTE((__unused__)),
-    void *const var_ptr, const void *const save) {
+int rocksdb_check_bulk_load_allow_unsorted(
+    THD *const thd, struct st_mysql_sys_var *var MY_ATTRIBUTE((__unused__)),
+    void *save, struct st_mysql_value *value) {
+  my_bool new_value;
+  int new_value_type = value->value_type(value);
+  if (new_value_type == MYSQL_VALUE_TYPE_STRING) {
```

Comment:
This is duplicated code - make a helper function to extract a my_bool from a st_mysql_value?

Reply:
Good point, I'll refactor this shortly.

Reply:
When I did the first fix for rocksdb_bulk_load I tried to find some place in sys_vars where this transformation of a bool-string value to a real bool but could not find anything that made sense, thus duplicating the server behavior that transforms "true", "false", "on", "off" to a true bool. It would be nice if we could somehow come up with a change to sys_var functionality that did this in some kind of a 'natural' way. I do not know if this is the first time this has come up but I am pretty sure it will not be the last.